src/oom/oomd-util.c

```diff
@@ -160,11 +160,13 @@ int oomd_fetch_cgroup_oom_preference(OomdCGroupContext *ctx, const char *prefix)
         if (r < 0)
                 return log_debug_errno(r, "Failed to get owner/group from %s: %m", ctx->path);
 
-        r = cg_get_owner(SYSTEMD_CGROUP_CONTROLLER, prefix, &prefix_uid);
-        if (r < 0)
-                return log_debug_errno(r, "Failed to get owner/group from %s: %m", ctx->path);
+        if (uid != 0) {
+                r = cg_get_owner(SYSTEMD_CGROUP_CONTROLLER, prefix, &prefix_uid);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to get owner/group from %s: %m", ctx->path);
+        }
```

Comment:
Maybe, better to check `uid == prefix_uid` here, and set `MANAGED_OOM_PREFERENCE_NONE` if the condition does not match:
```c
if (uid != 0) {
    uid_t prefix_uid;

    r = cg_get_owner(..., &prefix_uid);
    if (r < 0)
        return log_debug_errno(r, "...");

    if (prefix_uid != uid) {
        ctx->preference = MANAGED_OOM_PREFERENCE_NONE;
        return 0;
    }
}

r = cg_get_xattr_bool(...);
...
```

Reply:
Thanks, I agree this is better.